api.md

```diff
@@ -0,0 +1,115 @@
+# cozysdk-client
+
+SDK for serverless cozy apps
+
+## What is it for ?
+
+cozysdk-client is a native library that enables applications without a server to do requests directly, and simply to the data-system. 
+
+## What do we have inside ?
+
+To work with this library, you'll need to use the following functions :
+
+### create(docType, attributes)
+
+This ables the user to add a new item.
+
+#### What is docType ?
+
+You need to enter the docType, so for example if you want to define a request to get all the contacts, you'll have to add "Contacts" as a docType name. You also need to make sure that you added the permissions in the package.json like so : 
+  	
+  	"cozy-permissions": {
+    	"Contact": {
+      		"description": "To easily find your contact when talking about someone." 
+    	}
+  	}
+
+This is just an example but if you want to work with Emails for example, you'll just need to add "Message" as a cozy-premission and call it as a docType in the define function.
+
+#### What is attributes ?
+
+Attributes is the json object with the new fields that is added in the database. So for example if i want to add a user contact with bob as a name, I need to add a json object as an attribute as followed :
+
+	{"n": "bob"}
+
+#### A simple example
+
+	create("Contact", {"n": "bob"}, function(response) {
+		// You can get the id with response.id
+	});
+
+#### What does it return ?
+
+The response of this request will be the id of the new field that had been added.
+
+### find(id)
+
+This ables the user to get the data if a specific id.
+
+#### What does it return
+
+The response of this request will be a json object with the data of the id that had been used.
+
+### exists(id)
+
+This ables the user to know if the data of a specific id exists in the database.
+
+### updateAttributes(docType, id, attributes)
+
+This ables the user to update an attributes.
+
+### destroy(id)
+
+This ables the user to delete a specific field from the database.
+
+### defineRequest(docType, name, request)
+
+This ables the user to define a request. It's a case of overengineering that can change the behavior of cozysdk-client. Therefore, the developers focus on what computations need to be performed, as opposed to how those computations are actually carried out or how to get the data to the processes. So if the user wants to use a new docType, he will have to define the requests of this docType. 
+
+##### What is name ?
+
+The second param is the name of the request you want to create. So for example, if you want to get a contact that starts with an "a", you just need to create a name like "contactthatstartswithana" or whatsoever to be able to run it afterwards.
```

Comment:
It should be nice to have some conventions for requests/views. Maybe just a link to https://ehealthafrica.github.io/couchdb-best-practices/#naming-conventions-for-views